pkg/operator/k8sutil/node.go

```diff
@@ -157,3 +123,133 @@ func GetNodeSchedulable(node v1.Node) bool {
 	}
 	return true
 }
+
+// NodeMeetsPlacementTerms returns true if the Rook placement allows the node to have resources scheduled
+// on it. A node is placeable if it (1) meets any affinity terms that may be set in the placement,
+// and (2) its taints are tolerated by the placements tolerations.
+// There is the option to ignore well known taints defined in WellKnownTaints. See WellKnownTaints
```

Comment:
The WellKnownTaints are ignored when determining whether an OSD should be removed?

Reply:
Yes. The well-known taints are set when certain things are triggered on nodes. Like memory pressure, which causes `NoSchedule`. Or the node becoming not ready, which causes `NoExecute`. These are all things that are likely to be temporary, or at any rate, a circumstance where the user should be able to make the decision themselves whether the node should be cleaned of osds.